internal/admission/validator.go

```diff
@@ -630,3 +632,55 @@ func (m *managerClientConsumerGetter) ListAllConsumers(ctx context.Context) ([]k
 	}
 	return consumers.Items, nil
 }
+
+func (validator KongHTTPValidator) ValidateCustomEntity(ctx context.Context, entity kongv1alpha1.KongCustomEntity) (bool, string, error) {
+	// If the spec.contollerName does not match the ingress class name,
+	// ignore it as it is not controlled by the controller.
+	// REVIEW: should we also match the "ingress class" annotation here?
+	if !validator.ingressClassMatcher(&metav1.ObjectMeta{
+		Annotations: map[string]string{
+			annotations.IngressClassKey: entity.Spec.ControllerName,
+		},
+	}, annotations.IngressClassKey, annotations.ExactClassMatch) {
+		validator.Logger.V(util.DebugLevel).Info("Skipped validation because the controller name does not match", "controller_name", entity.Spec.ControllerName)
+		return true, "", nil
+	}
+
+	// REVIEW: should we validate group/kind of `spec.parentRef` here because we only want it to be attached to plugins?
```

Comment:
I don't think we need it. We're intentionally leaving it open to possibly supporting other kinds in the future. For the time being docs are only going to instruct you to use plugins here, and we don't really need warnings for whatever tiny set of users try to use something else out of sheer curiosity.

Reply:
Done.